test/README.md

```diff
@@ -0,0 +1,125 @@
+# Testing Guidelines #
+
+## Rationale ##
+
+Testing is a necessary aspect of any software development process. There are
+many ways to describe the quality of a test suite (both quantitative and
+qualitative). In addition to these properties, a high quality test suite still
+necessitates clear and coherent organisation.
+
+Whilst the test suite for Perpetual Pools exhibits excellent coverage metrics
+and captures the intended behaviour of the system rather well, it is currently
+organised very poorly (inconsistent naming, inconsistent file structure,
+inconsistent use of testing directives, etc.).
+
+This document aims to clarify how we conduct testing in the Perpetual Pools
+system and act as a procedural guide for how to write additional tests when
+contributing to the codebase. Specifically, this document aims to address the
+following questions:
+
+ - How should tests be laid out? File-per-contract, etc.
+ - What testing frameworks do we use?
+ - How should the tests themselves look?
+
+## Test Suite Structure ##
+
+### All Tests Under `test` ###
+
+All tests -- regardless of their type (e.g., unit, integration, etc.) -- must
+reside under the `test` directory.
+
+### One-File-One-Contract ###
+
+All unit tests should be arranged such that all tests for a given module are in
+the same file. The most obvious choice for what constitutes a "module" is a
```

Comment:
Do you mean under the same folder? Right now each contract is under a folder with multiple files testing a different aspect of it. Would be chaotic to have em all one file.

Reply:
> Would be chaotic to have em all one file

I don't really see how? If by "chaotic" you mean the files would get "large", I don't necessarily think the size of the file is an issue -- **provided the tests adhere to consistent structure**.

My personal preference is to group tests by the *code* they test. An alternative approach would be to group them by the *functionality* they test. I'm not against the latter, but whatever we decided on as a team, we need to stick with. Currently, our test suite uses both approaches, which definitely *is* chaotic.

Reply:
I think very large files when writing tests gets very annoying to deal with. I would want to push more for one folder per module and one file per functionality as you mentioned @sporejack 

I think it gets annoying when working with large test files mostly when you are having to move between different tests, or look for certain parts of the test file, etc. I think another layer of indexing here would just make it a bit nicer.

Reply:
Consensus seems pretty clear on this one -- I'll remove this section.

Reply:
Thoughts now @mynameuhh ?

Reply:
Yep looks good @sporejack